In the tests for https://github.com/galaxyproject/galaxy/pull/9425/commits/47ae6b2816b6ee73aa76cbe5bb8e90505ec2c267 I have seen `'__Infinity__'` and a good explanation seemed to be that the test description is encoded in json at some point which introduces this string. 

For instance in the converter tests https://github.com/galaxyproject/galaxy/runs/643979914:

```
======================================================================
FAIL: ( CONVERTER_interval_to_bigwig_0 ) > Test-1
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/runner/work/galaxy/galaxy/galaxy root/test/functional/test_toolbox.py", line 99, in test_tool
    self.do_it(tool_version=tool_version, test_index=test_index)
  File "/home/runner/work/galaxy/galaxy/galaxy root/test/functional/test_toolbox.py", line 36, in do_it
    verify_tool(tool_id, self.galaxy_interactor, resource_parameters=resource_parameters, test_index=test_index, tool_version=tool_version, register_job_data=register_job_data)
  File "/home/runner/work/galaxy/galaxy/galaxy root/lib/galaxy/tool_util/verify/interactor.py", line 852, in verify_tool
    raise e
  File "/home/runner/work/galaxy/galaxy/galaxy root/lib/galaxy/tool_util/verify/interactor.py", line 848, in verify_tool
    job_stdio = _verify_outputs(testdef, test_history, jobs, tool_id, data_list, data_collection_list, galaxy_interactor, quiet=quiet)
  File "/home/runner/work/galaxy/galaxy/galaxy root/lib/galaxy/tool_util/verify/interactor.py", line 1023, in _verify_outputs
    raise JobOutputsError(found_exceptions, job_stdio)
galaxy.tool_util.verify.interactor.JobOutputsError: could not convert string to float: '__Infinity__'
```

I did reset to this commit for demonstration purposes (for by own reference the sha of the other commit: 8f72dfe8ee).

> Is that really necessary for this PR ?

No, we just need a good default value for `delta_frac`. `Inf` would have been perfect in my opinion since it would accept any difference -- which is what is needed. Maybe `None` or a negative value which we would need to test for to skip the `delta_frac` test ...

--- a/lib/galaxy/tool_util/parser/util.py
+++ b/lib/galaxy/tool_util/parser/util.py
@@ -1,6 +1,7 @@
 from collections import OrderedDict
 
-DEFAULT_DELTA = "10000"
+DEFAULT_DELTA = 10000
+DEFAULT_DELTA_FRAC = None
 
 
 def is_dict(item):

--- a/lib/galaxy/tool_util/parser/xml.py
+++ b/lib/galaxy/tool_util/parser/xml.py
@@ -9,7 +9,10 @@ from math import isinf
 import packaging.version
 
 from galaxy.tool_util.deps import requirements
-from galaxy.tool_util.parser.util import DEFAULT_DELTA
+from galaxy.tool_util.parser.util import (
+    DEFAULT_DELTA,
+    DEFAULT_DELTA_FRAC
+)
 from galaxy.util import string_as_bool, xml_text, xml_to_string
 from .interface import (
     InputSource,
@@ -614,6 +617,7 @@ def __parse_test_attributes(output_elem, attrib, parse_elements=False, parse_dis
     attributes['lines_diff'] = int(attrib.pop('lines_diff', '0'))
     # Allow a file size to vary if sim_size compare
     attributes['delta'] = int(attrib.pop('delta', DEFAULT_DELTA))
+    attributes['delta_frac'] = float(attrib['delta_frac']) if 'delta_frac' in attrib else DEFAULT_DELTA_FRAC
     attributes['sort'] = string_as_bool(attrib.pop('sort', False))
     attributes['decompress'] = string_as_bool(attrib.pop('decompress', False))
     extra_files = []

--- a/lib/galaxy/tool_util/parser/yaml.py
+++ b/lib/galaxy/tool_util/parser/yaml.py
@@ -3,7 +3,10 @@ from collections import OrderedDict
 import packaging.version
 
 from galaxy.tool_util.deps import requirements
-from galaxy.tool_util.parser.util import DEFAULT_DELTA
+from galaxy.tool_util.parser.util import (
+    DEFAULT_DELTA,
+    DEFAULT_DELTA_FRAC
+)
 from .interface import (
     InputSource,
     PageSource,
@@ -219,7 +222,8 @@ def _parse_test(i, test_dict):
         defaults = {
             'compare': 'diff',
             'lines_diff': 0,
-            'delta': int(DEFAULT_DELTA),
+            'delta': DEFAULT_DELTA,
+            'delta_frac': DEFAULT_DELTA_FRAC,
             'sort': False,
         }
         # TODO

--- a/lib/galaxy/tool_util/verify/__init__.py
+++ b/lib/galaxy/tool_util/verify/__init__.py
@@ -16,7 +16,10 @@ try:
 except ImportError:
     pysam = None
 
-from galaxy.tool_util.parser.util import DEFAULT_DELTA
+from galaxy.tool_util.parser.util import (
+    DEFAULT_DELTA,
+    DEFAULT_DELTA_FRAC
+)
 from galaxy.util import unicodify
 from galaxy.util.compression_utils import get_fileobj
 from .asserts import verify_assertions
@@ -132,11 +135,7 @@ def verify(
             elif compare == 're_match_multiline':
                 files_re_match_multiline(local_name, temp_name, attributes=attributes)
             elif compare == 'sim_size':
-                delta = int(attributes.get('delta', DEFAULT_DELTA))
-                s1 = len(output_content)
-                s2 = os.path.getsize(local_name)
-                if abs(s1 - s2) > int(delta):
-                    raise AssertionError('Files %s=%db but %s=%db - compare by size (delta=%s) failed' % (temp_name, s1, local_name, s2, delta))
+                files_delta(local_name, temp_name, attributes=attributes)
             elif compare == "contains":
                 files_contains(local_name, temp_name, attributes=attributes)
             else:
@@ -192,6 +191,20 @@ def _verify_checksum(data, checksum_type, expected_checksum_value):
         template = "Output checksum [%s] does not match expected [%s] (using hash algorithm %s)."
         message = template % (actual_checksum_value, expected_checksum_value, checksum_type)
         raise AssertionError(message)
+
+
+def files_delta(file1, file2, attributes=None):
+    """Check the contents of 2 files for size differences."""
+    if attributes is None:
+        attributes = {}
+    delta = attributes.get('delta', DEFAULT_DELTA)
+    delta_frac = attributes.get('delta_frac', DEFAULT_DELTA_FRAC)
+    s1 = os.path.getsize(file1)
+    s2 = os.path.getsize(file2)
+    if abs(s1 - s2) > delta:
+        raise AssertionError('Files %s=%db but %s=%db - compare by size (delta=%s) failed' % (file1, s1, file2, s2, delta))
+    if delta_frac is not None and not (s1 - (s1 * delta_frac) <= s2 <= s1 + (s1 * delta_frac)):
+        raise AssertionError('Files %s=%db but %s=%db - compare by size (delta_frac=%s) failed' % (file1, s1, file2, s2, delta_frac))
 
 
 def files_diff(file1, file2, attributes=None):

--- a/lib/galaxy/tool_util/xsd/galaxy.xsd
+++ b/lib/galaxy/tool_util/xsd/galaxy.xsd
@@ -1312,9 +1312,14 @@ When this attribute is true and ``compare`` is set to ``diff``, try to decompres
 ]]></xs:documentation>
       </xs:annotation>
     </xs:attribute>
-    <xs:attribute name="delta" type="xs:integer">
+    <xs:attribute name="delta" type="xs:integer" default="10000">
       <xs:annotation>
-        <xs:documentation xml:lang="en">If ``compare`` is set to ``sim_size``, this is the number of bytes different allowed (default 10000).</xs:documentation>
+        <xs:documentation xml:lang="en">If ``compare`` is set to ``sim_size``, this is the maximum allowed absolute size difference (in bytes) between the data set that is generated in the test and the file in ``test-data/`` that is referenced by the ``file`` attribute. Can be combined with ``delta_frac``.</xs:documentation>
+      </xs:annotation>
+    </xs:attribute>
+    <xs:attribute name="delta_frac" type="xs:float">
+      <xs:annotation>
+        <xs:documentation xml:lang="en">If ``compare`` is set to ``sim_size``, this is the maximum allowed relative size difference between the data set that is generated in the test and the file in ``test-data/`` that is referenced by the ``file`` attribute. A value of 0.1 means that the file that is generated in the test can differ by at most 10% of the file in ``test-data``. The default is not to check for  relative size difference. Can be combined with ``delta``.</xs:documentation>
       </xs:annotation>
     </xs:attribute>
   </xs:complexType>

--- a/test/functional/tools/samples_tool_conf.xml
+++ b/test/functional/tools/samples_tool_conf.xml
@@ -55,6 +55,7 @@
   <tool file="qc_stdout.xml" />
   <tool file="md5sum.xml" />
   <tool file="checksum.xml" />
+  <tool file="sim_size_delta.xml" />
   <tool file="composite_shapefile.xml" />
   <tool file="is_valid_xml.xml" />
   <tool file="param_value_from_file.xml"/>

--- a/test/functional/tools/sim_size_delta.xml
+++ b/test/functional/tools/sim_size_delta.xml
@@ -1,0 +1,22 @@
+<tool id="sim_size_delta" name="sim_size_delta" version="0.1.0">
+  <command><![CDATA[
+    head -c 80 '$in' > '$out_smaller' &&
+    cp '$in' '$out_larger' &&
+    head -c 20 '$in' >> '$out_larger'
+  ]]></command>
+  <inputs>
+    <param name="in" type="data" format="data"/>
+  </inputs>
+  <outputs>
+    <data name="out_smaller" format="data" label="smaller"/>
+    <data name="out_larger" format="data" label="larger"/>
+  </outputs>
+  <tests>
+    <!-- take a test file of 100b size and add/remove 20b -->
+    <test expect_num_outputs="2">
+      <param name="in" value="1.sff" ftype="txt" />
+      <output name="out_smaller" value="1.sff" compare="sim_size" delta="20" delta_frac="0.2"/>
+      <output name="out_larger" value="1.sff" compare="sim_size" delta="20" delta_frac="0.2"/>
+    </test>
+  </tests>
+</tool>

--- a/test/unit/test_verify.py
+++ b/test/unit/test_verify.py
@@ -6,6 +6,7 @@ import pytest
 
 from galaxy.tool_util.verify import (
     files_contains,
+    files_delta,
     files_diff,
     files_re_match,
     files_re_match_multiline,
@@ -20,14 +21,18 @@ MULTILINE_MATCH = b".*"
 TestFile = collections.namedtuple('TestFile', 'value path')
 
 
-def generate_tests(multiline=False):
+def test_file_list():
     files = []
     for b, ext in [(F1, '.txt'), (F2, '.txt'), (F3, '.pdf'), (F4, '.txt'), (MULTILINE_MATCH, '.txt')]:
         fd, path = tempfile.mkstemp(suffix=ext)
         with os.fdopen(fd, 'wb') as out:
             out.write(b)
         files.append(TestFile(b, path))
-    f1, f2, f3, f4, multiline_match = files
+    return files
+
+
+def generate_tests(multiline=False):
+    f1, f2, f3, f4, multiline_match = test_file_list()
     if multiline:
         tests = [(multiline_match, f1, {'lines_diff': 0, 'sort': True}, None)]
     else:
@@ -40,6 +45,23 @@ def generate_tests(multiline=False):
     return tests
 
 
+def generate_tests_sim_size():
+    f1, f2, f3, f4, multiline_match = test_file_list()
+    # tests for equal files
+    tests = [(f1, f1, None, None),  # pass default values
+             (f1, f1, {'delta': 0}, None),  # pass for values that should always pass
+             (f1, f1, {'delta_frac': 0.0}, None)]
+    # tests for two different file (diff is 422 lines, 0.011709602)
+    tests += [(f1, f2, None, None),  # pass default values
+             (f1, f2, {'delta': 422}, None),  # test around the actual difference of two different
+             (f1, f2, {'delta_frac': 84.4}, None),
+             (f1, f2, {'delta': 421}, AssertionError),
+             (f1, f2, {'delta_frac': 84.3}, AssertionError),
+             (f1, f2, {'delta': 422, 'delta_frac': 84.3}, AssertionError),  # test with combination where at least one leads to an assertion
+             (f1, f2, {'delta': 421, 'delta_frac': 84.4}, AssertionError)]
+    return tests
+
+
 @pytest.mark.parametrize('file1,file2,attributes,expect', generate_tests())
 def test_files_contains(file1, file2, attributes, expect):
     if expect is not None:
@@ -47,6 +69,15 @@ def test_files_contains(file1, file2, attributes, expect):
             files_contains(file1.path, file2.path, attributes)
     else:
         files_contains(file2.path, file2.path, attributes)
+
+
+@pytest.mark.parametrize('file1,file2,attributes,expect', generate_tests_sim_size())
+def test_files_delta(file1, file2, attributes, expect):
+    if expect is not None:
+        with pytest.raises(expect):
+            files_delta(file1.path, file2.path, attributes)
+    else:
+        files_delta(file1.path, file2.path, attributes)
 
 
 @pytest.mark.parametrize('file1,file2,attributes,expect', generate_tests())

